storage/remote/write.go

```diff
@@ -88,20 +87,17 @@ func (rws *WriteStorage) ApplyConfig(conf *config.Config) error {
 	rws.mtx.Lock()
 	defer rws.mtx.Unlock()
 
-	// Remote write queues only need to change if the remote write config or
-	// external labels change. Hash these together and only reload if the hash
-	// changes.
-	cfgBytes, err := json.Marshal(conf.RemoteWriteConfigs)
+	configHash, err := toHash(conf.RemoteWriteConfigs)
```

Comment:
Is this necessary anymore or should we just let the loop go through a few times?

Reply:
It's a pretty cheap short circuit to doing the rest of the loop, which is also pretty cheap. I'd keep it personally, but move it to `storage.go`'s `ApplyConfig` and include a similar check for the remote read config.

Reply:
:point_up: This can be it's own PR